week-06-intro-ruby-on-rails/day-03/module-01/readme.md

```diff
@@ -1 +1,244 @@
+
+![Class inheritance](http://i.stack.imgur.com/rvcEi.png)
+
+## Classes
+
+<details>
+  <summary>How do we create a class named Car that goes "Putt-Putt" when it's initialized?</summary>
+
+  ```ruby
+  class Car
+    def initialize
+      p "Putt-Putt"
+    end
+  end
+  ```
+</details>
+
+## Attributes
+
+```ruby
+fiat = Car.new
+fiat.color = "hot pink"
+fiat.color
+=> "hot pink"
+```
+
+<details>
+  <summary>What should we do if we want to set attributes on the car, such as a paint color and year?</summary>
+
+  ```ruby
+  class Car
+    attr_accessor :color
+  end
+  ```
+</details>
+
+## Methods
+
+Do you remember the difference between methods and functions in JavaScript?
+
+Methods ending in `?` are conventionally used to denote that they return a boolean value. <a href="https://github.com/bbatsov/ruby-style-guide#naming" target="_blank">Ruby Style Guide</a>.
+
+<details>
+  <summary>How would we create an instance method `color?` for our `Car` class instances, which tests whether the color matches what is passed in?</summary>
+
+  ```ruby
+  class Car
+    attr_accessor :color
+
+    def color?(other_color)
+      return color == other_color
+    end
+  end
+  ```
+</details>
+
+```ruby
+# Example call.
+alpha = Car.new
+alpha.color = "orange"
+alpha.color?("blue")
+=> false
+alpha.color?("orange")
+=> true
+```
+
+## Initialization and Instance Variables
+
+<details>
+  <summary>What if I want to set the color of the car at the moment I'm creating it? How could I enable code like this?
+
+  ```ruby
+  lambo = Car.new("red")
+  lambo.color
+  => "red"
+  ```
+  </summary>
+
+  ```ruby
+  class Car
+    attr_reader :color
+
+    def initialize(color)
+      @color = color
+    end
+  end
+  ```
+</details>
+
+## Class Variables
+
+<details>
+  <summary>What if I wanted a running counter for all the cars I've ever created? Enable the code: `Car.total` which outputs the total number of all cars ever created.</summary>
+
+  ```ruby
+  class Car
+    @@total = 0
+
+    def initialize
+      @@total += 1
+    end
+
+    def self.total
+      @@total
+    end
+  end
+  ```
+</details>
+
+## Quick Review
+
+Think, pair, share...
+
+  * What is a class?
+    - What is an attribute?
+    - What is a method?
+  * What is the difference between:
+    - an instance variable.
+    - a class variable.
+  * Why do we use classes?
+  * What is inheritance?
+
+## Inheritance
+
+Given an `Vehicle` class that contains the method `accelerate` and attribute `speed`..
+
+```ruby
+class Vehicle
+  attr_accessor :speed
+  def initialize
+    @speed = 0
+  end
+
+  def accelerate(speed)
+    @speed += speed
+  end
+end
+```
+
+<details>
+  <summary>How can I create a `Car` class while being DRY and not duplicating the method `accelerate`? Let's inherit from the `Vehicle` class above and make the car hella fast.</summary>
+
+  ```ruby
+  class Car < Vehicle
+    def initialize
+      @speed = 2
+    end
+
+    def accelerate(speed)
+      @speed *= speed
```

Comment:
Might be a silly question, but is *= an operator I'm not familiar with, or is this a typo?